### ISSUE TITLE
Random Javascript Error

### ISSUE THREAD

**Issue by taylorwastaken** [aza547/wow-recorder#199]
Just keep getting random javascript errors while running keys.

[WarcraftRecorder-2022-10-09.log](https://github.com/aza547/wow-recorder/files/9743046/WarcraftRecorder-2022-10-09.log)

**Comment by aza547** [comment]
Could you also share the OBS logs? In a folder at the location the log button in the app takes you to.

**Comment by taylorwastaken** [comment]
[2022-10-09 18-02-35.txt](https://github.com/aza547/wow-recorder/files/9745744/2022-10-09.18-02-35.txt)
[2022-10-09 21-59-56.txt](https://github.com/aza547/wow-recorder/files/9745745/2022-10-09.21-59-56.txt)

**Comment by taylorwastaken** [comment]
![javascript2](https://user-images.githubusercontent.com/65497335/194903815-e4de0f16-15f4-46db-9e87-98f547e8c6da.PNG)
![javascript3](https://user-images.githubusercontent.com/65497335/194903818-3b1c8a75-d8da-4687-84a2-84d1ed54f72d.PNG)
![javascript1](https://user-images.githubusercontent.com/65497335/194903819-81b7a902-8bfc-43b2-a85c-8ac4bd545ef1.PNG)


here are the three errors that pop up. I was not doing any key just had wow open and then I let it disconnect from time out.

**Comment by taylorwastaken** [comment]
[2022-10-10 08-02-31.txt](https://github.com/aza547/wow-recorder/files/9747287/2022-10-10.08-02-31.txt)
heres the OBS log associated with those images

**Comment by aza547** [comment]
Might be the same as #194.

What is the type disk you are saving to?

**Comment by aza547** [comment]
Almost certain it's the same issue.

13 minutes to cut the video in one instance. 

Looks like you started another key before it was done cutting. 

We have fixes in the work for this, hopefully available next weekend sorry you have hit this.

If you have a faster disk you can use meanwhile that might help.

**Comment by taylorwastaken** [comment]
its saving to a regular HDD, ill change to have it save to my SSD. Thanks

**Comment by aza547** [comment]
Closing as a duplicate of #194

**Comment by taylorwastaken** [comment]
just adding more to this. I moved my recordings to the ssd and still received similar errors
[WarcraftRecorder-2022-10-10.log](https://github.com/aza547/wow-recorder/files/9757349/WarcraftRecorder-2022-10-10.log)
[WarcraftRecorder-2022-10-11.log](https://github.com/aza547/wow-recorder/files/9757350/WarcraftRecorder-2022-10-11.log)
[2022-10-10 22-03-53.txt](https://github.com/aza547/wow-recorder/files/9757351/2022-10-10.22-03-53.txt)
[2022-10-11 08-10-46.txt](https://github.com/aza547/wow-recorder/files/9757352/2022-10-11.08-10-46.txt)
![javascript5](https://user-images.githubusercontent.com/65497335/195128423-0b2354eb-fb88-4869-b385-a8484062b232.PNG)
![javascript6](https://user-images.githubusercontent.com/65497335/195128430-39da17de-66df-4d09-8801-4489d2ac71f4.PNG)
![javascript4](https://user-images.githubusercontent.com/65497335/195128433-1530f8c0-e335-4cbf-b783-83700b91cf80.PNG)

**Comment by taylorwastaken** [comment]
I achieved this by just pressing the record button and then stopping it. and then trying to record again.

**Comment by aza547** [comment]
Well that's interesting...
The move to SSD seems to have worked and sped up the cutting. 

```
[2022-10-11 11:02:01.933] [info]  [Util] Ready to cut video.
[2022-10-11 11:02:03.053] [info]  [Util] FFmpeg cut video succeeded
```

That took 2 seconds, which is within what I expect. 

So this looks like something new.

**Comment by aza547** [comment]
I can see you've spammed your test button a bit - can you retry this by restarting the app and clicking the test button only once? I think this might be the cause of your problem. Although we have a guard meant to prevent this happening, it's clearly not working! I've fixed that here:
- https://github.com/aza547/wow-recorder/commit/89c2c388653159eabdd94411417b8c2c628228d8
- https://github.com/aza547/wow-recorder/commit/1aef1b061dcd3038f1782b2fb27eef768ed6927c

**Comment by taylorwastaken** [comment]
looks like that was it, i couldnt tell if it had started so i clicked it alot. but yea i just waited a bit and go it no errors. thanks again!

**Comment by aza547** [comment]
Great, no worries! Thanks for the report!

**Comment by taylorwastaken** [comment]
i hate to keep reopening this but it happened again and i wasnt even in a key or anything just launching the game from beta back to live. same style javascript errors as posted above
[WarcraftRecorder-2022-10-12.log](https://github.com/aza547/wow-recorder/files/9760502/WarcraftRecorder-2022-10-12.log)
[2022-10-11 20-13-53.txt](https://github.com/aza547/wow-recorder/files/9760503/2022-10-11.20-13-53.txt)

**Comment by aza547** [comment]
No need to feel guilty, bug reports like this are super helpful! I'll look shortly.

**Comment by aza547** [comment]
Strange one this time. Again different than the last. I need some more time for this, but meanwhile you should take 2.9.0. There were some fixes in this area and it might just go away.

**Comment by aza547** [comment]
Are you still having issues here? @taylorwastaken

**Comment by taylorwastaken** [comment]
Not since I updated to the latest and swapped to the ssd. Thanks! I didn’t realize this was still open my bad.
